src/webui/www/private/scripts/misc.js

```diff
@@ -167,3 +167,37 @@ function toFixedPointString(number, digits) {
     const power = Math.pow(10, digits);
     return (Math.floor(power * number) / power).toFixed(digits);
 }
+
+/**
+ *
+ * @param {String} text the text to search
+ * @param {Array<String>} terms terms to search for within the text
+ * @returns {Boolean} true if all terms match the text, false otherwise
+ */
+function containsAllTerms(text, terms) {
+    const textToSearch = text.toLowerCase();
+    for (let i = 0; i < terms.length; ++i) {
+        const term = terms[i].trim().toLowerCase();
+        if ((term[0] === '-')) {
```

Comment:
What does happen if "term" is empty here?

Reply:
If `term` is empty then `term[0]` will return `undefined`. This will result in none of the conditionals being satisfied (`value.indexOf("")` equals `0`).